tests/framework/base/SecurityTest.php

```diff
@@ -1219,43 +1219,6 @@ public function testHkdf($hash, $ikm, $salt, $info, $l, $prk, $okm)
         $this->assertEquals($okm, bin2hex($dk));
     }
 
-    public function dataProviderCompareStrings()
-    {
-        return [
-            ['', ''],
-            [false, ''],
-            [null, ''],
-            [0, ''],
-            [0.00, ''],
-            ['', null],
-            ['', false],
-            ['', 0],
-            ['', "\0"],
-            ["\0", ''],
-            ["\0", "\0"],
-            ['0', "\0"],
-            [0, "\0"],
-            ['user', 'User'],
-            ['password', 'password'],
-            ['password', 'passwordpassword'],
-            ['password1', 'password'],
-            ['password', 'password2'],
-            ['', 'password'],
-            ['password', ''],
-        ];
-    }
-
-    /**
-     * @dataProvider dataProviderCompareStrings
-     *
-     * @param $expected
-     * @param $actual
-     */
-    public function testCompareStrings($expected, $actual)
-    {
-        $this->assertEquals(strcmp($expected, $actual) === 0, $this->security->compareString($expected, $actual));
```

Comment:
I'd keep it to provide complete interface, we also have password hash method that just wraps PHP now.